### ISSUE TITLE
Feature Request: Link Live Preview to Content Editor

### ISSUE THREAD

**Issue by RitterKnightCreative** [craftcms/cms#2811]
Raise your hand if you've ever started editing a piece of content, did a live preview on it, found a typo and then tried to click to edit it.  Especially with the power of Matrix, you can end up having to wade back through a dozen blocks to see where exactly to edit it.

Live Preview is awesome. I think it's the best blend between keeping content structured and allowing editors to see what it's going to look like. 

I'm also not convinced that an "inline" style editor is the way to go but it would be great to be able to click on an element in Live Preview and then Craft editing pane would jump you to/blink where it might be since Live Preview is basically a one-way street right now. 

Also, bonus points if you highlighted text and that particular text was then highlighted in the editor, ready to be edited.

The way I envision this working might be that developers could "tag" (in Twig) which field/matrix block they wanted target. Craft would add whatever click triggers would be necessary in Live Preview and then when someone clicks to engage Live Preview, the editor would focus on/blink that particular field/matrix block... eg:

     {% livePreview 'fieldHandle.matrixBlockField' %}
      <div>whatever markup</div>
    {% endlivePreview %}

That would also allow devs to opt-in or out of this functionality instead of having Craft try to detect what the right thing to do would be.

It'd obviously be more complex than that but without going full blown inline editing as some CMS are going, I think this could be a reasonable middle ground. Personally I think CMS specific markup needs to stay out of the HTML code which is where Twig could really shine here because it would only insert the necessary html/JS triggers only in live preview.

**Comment by narration-sd** [comment]
@RitterKnightCreative  that's an entertaining idea, and probably gives more food for thought in Bend Central than I'm able to respond to. A little imagining  on how this could work might  make it  easier to have compatible with future headless Vuejs etc./Live Preview editing, as I am preparing to make easy, though. 

To get back to the Craft (matrix) editor, you'd need a JS call to the parent of the iframe the live preview lives in. What could be convenient could be an identity code for each element, rather than requiring composition of a matching object tree pathname -- though it could be that this.that.included pathname.

If it were part of the data Live Preview passes in its post to the preview template, it could be picked off for the Vuejs etc. app conveniently, and as easily used for the Twig rendition normally. Which could give the functionality you ask for without having to code embracing Twig. I'm imagining it would turn up in a data-* attribute on each element required -- and could do the same on the Craft editor side, making an easy match for the callback.

Anyway,  moment's thought towards this...,

**Comment by brandonkelly** [comment]
Yeah we’ve thought about doing something like this – at least so Craft has an idea of where on the page the various fields are being output, so that it can scroll the panes together.
